addons/test_mail_full/tests/test_portal.py

```diff
@@ -96,6 +96,40 @@ def test_portal_avatar_image(self):
         self.assertEqual(no_token_response.headers.get('Content-Type'), 'image/png')
         self.assertRegex(no_token_response.headers.get('Content-Disposition', ''), r'placeholder\.png')
 
+    def test_portal_avatar_with_hash_pid(self):
+        self.authenticate(None, None)
+        post_url = f"{self.record_portal.get_base_url()}/mail/chatter_post"
+        self.opener.post(
+            url=post_url,
+            json={
+                'params': {
+                    'csrf_token': http.Request.csrf_token(self),
+                    'message': 'Test',
+                    'res_model': self.record_portal._name,
+                    'res_id': self.record_portal.id,
+                    'hash': self.record_portal._sign_token(self.partner_2.id),
+                    'pid': self.partner_2.id,
+                },
+            },
+        )
```

Comment:
I don't get the purpose of this call

Reply:
To send a message on portal chatter with `hash and pid` when there is no access token.

Reply:
Then please add the following:
```py
res = self.opener.post(...)
res.raise_for_status()
self.assertNotIn("error", res.json())```

Reply:
Done